daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/IncompleteTransaction.scala

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf
+package transaction
+
+import com.daml.lf.value.Value
+
+trait IncompleteTransaction {
+
+  type Nid = NodeId
+  type Cid = Value.ContractId
+  type TX = GenTransaction[Nid, Cid]
+  type ExerciseNode = Node.NodeExercises[Nid, Cid]
+
+  def transaction: TX
+  def exerciseContextMaybe: Option[ExerciseNode]
+}
```

Comment:
I think  it will be convenient that an `IncompleteTransaction` is just a normal transaction like we have done for `SubmittedTransaction` and `CommitedTransaction`. 

You could add in package.scala

```
  val IncompleteTransaction = DiscriminatedSubtype[VersionedTransaction[NodeId, ContractId]]
  type IncompleteTransaction = IncompleteTransaction.T
```

Reply:
this is a suggestion.